modules/move-etcd-different-disk.adoc

```diff
@@ -50,146 +82,91 @@ metadata:
-          DefaultDependencies=no
-          After=var-lib-etcd.mount var.mount
-          Before=crio.service
-
-          [Service]
-          Type=oneshot
-          RemainAfterExit=yes
-          ExecCondition=/usr/bin/test ! -d /var/lib/etcd/member
-          ExecStart=semanage fcontext -a -e /sysroot/ostree/deploy/rhcos/var/lib/etcd/ /var/lib/etcd/
-          ExecStart=/bin/rsync -ar /sysroot/ostree/deploy/rhcos/var/lib/etcd/ /var/lib/etcd/
-          TimeoutSec=0
-
-          [Install]
-          WantedBy=multi-user.target graphical.target
-        enabled: true
-        name: sync-var-lib-etcd-to-etcd.service
-      - contents: |
-          [Unit]
-          Description=Restore recursive SELinux security contexts
-          DefaultDependencies=no
-          After=var-lib-etcd.mount
-          Before=crio.service
-
-          [Service]
-          Type=oneshot
-          RemainAfterExit=yes
-          ExecStart=/sbin/restorecon -R /var/lib/etcd/
-          TimeoutSec=0
-
-          [Install]
-          WantedBy=multi-user.target graphical.target
-        enabled: true
-        name: restorecon-var-lib-etcd.service
+        - name: find-secondary-device.service
+          enabled: true
+          contents: |
+            [Unit]
+            Description=Find secondary device
+            DefaultDependencies=false
+            After=systemd-udev-settle.service
+            Before=local-fs-pre.target
+            ConditionPathExists=!/etc/var-lib-etcd-mount
+
+            [Service]
+            RemainAfterExit=yes
+            ExecStart=/etc/find-secondary-device
+
+            RestartForceExitStatus=77
+
+            [Install]
+            WantedBy=multi-user.target
+        - name: var-lib-etcd.mount
+          enabled: true
+          contents: |
+            [Unit]
+            Before=local-fs.target
+
+            [Mount]
+            What=/dev/disk/by-label/var-lib-etcd
+            Where=/var/lib/etcd
+            Type=xfs
+            TimeoutSec=120s
+
+            [Install]
+            RequiredBy=local-fs.target
+        - name: sync-var-lib-etcd-to-etcd.service
+          enabled: true
+          contents: |
+            [Unit]
+            Description=Sync etcd data if new mount is empty
+            DefaultDependencies=no
+            After=var-lib-etcd.mount var.mount
+            Before=crio.service
+
+            [Service]
+            Type=oneshot
+            RemainAfterExit=yes
+            ExecCondition=/usr/bin/test ! -d /var/lib/etcd/member
+            ExecStart=/usr/sbin/setsebool -P rsync_full_access 1
+            ExecStart=/bin/rsync -ar /sysroot/ostree/deploy/rhcos/var/lib/etcd/ /var/lib/etcd/
+            ExecStart=/usr/sbin/semanage fcontext -a -t container_var_lib_t '/var/lib/etcd(/.*)?'
+            ExecStart=/usr/sbin/setsebool -P rsync_full_access 0
+            TimeoutSec=0
+
+            [Install]
+            WantedBy=multi-user.target graphical.target
+        - name: restorecon-var-lib-etcd.service
```

Comment:
just 1 thing here, wouldn't this restorecon run every time ? if so, is this necessary? 
other than this, the rest looks good to me.

Reply:
> just 1 thing here, wouldn't this restorecon run every time ? if so, is this necessary? other than this, the rest looks good to me.

@kenneth-dsouza your thoughts?

Reply:
Yes, but it should not cause any harm. I have kept the same behavior as the previous releases.